libs/libc/unistd/lib_getcwd.c

```diff
@@ -115,6 +124,16 @@ FAR char *getcwd(FAR char *buf, size_t size)
       return NULL;
     }
 
+  if (buf == NULL)
+    {
+      buf = malloc(size);
```

Comment:
malloc() should not be used.  Use lib_malloc() instead.  That eliminates the problem case where this function is called from within the OS in certain build modes and malloc does not do what you would want it do do.

Reply:
Okay, thanks. Done!